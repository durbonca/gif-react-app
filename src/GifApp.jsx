import React, { useState } from 'react'
import {AddCategory} from './components/AddCategory'
import {GifGrid} from './components/GifGrid'

export const GifApp = () => {

    const [categories, setCategories] = useState(['Hola'])

    const handleAdd = ({ input }) => {
        setCategories([input, ...categories ])
    }

  return (
      <>
        <h2>Gif App</h2>
        <AddCategory handleAdd={handleAdd} />
        <hr />
        <ol>
            {categories.map(category => 
                <GifGrid key={category} category={category} />
            )}
        </ol>
      </>
  )
}
