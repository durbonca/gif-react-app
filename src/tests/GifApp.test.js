import React from 'react';
import { shallow } from 'enzyme';
import { GifApp } from '../GifApp';

describe('GifApp test', ()=>{
    test('return a component', ()=>{
        const component = shallow(<GifApp/>);
        expect(component).toMatchSnapshot();
        expect(component.find('h2').text().trim()).toBe('Gif App');
    })
})