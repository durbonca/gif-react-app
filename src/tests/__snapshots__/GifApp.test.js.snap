// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifApp test return a component 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <AddCategory
    handleAdd={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hola"
      key="Hola"
    />
  </ol>
  <div
    className="footer"
  >
    <div>
      made by 
      <a
        href="https://durbonca.com"
        rel="noreferrer"
        target="_blank"
      >
        @durbonca
      </a>
    </div>
    <div>
      Powered by 
      <a
        href="https://giphy.com/"
        rel="noreferrer"
        target="_blank"
      >
        Giphy
      </a>
    </div>
  </div>
</Fragment>
`;
